haystack/nodes/prompt/prompt_node.py

```diff
@@ -964,19 +967,121 @@ def run(
         final_result["_debug"] = {"prompts_used": prompt_collector}
         return final_result, "output_1"
 
-    def run_batch(
+    def run_batch(  # type: ignore
         self,
-        queries: Optional[Union[str, List[str]]] = None,
-        file_paths: Optional[List[str]] = None,
-        labels: Optional[Union[MultiLabel, List[MultiLabel]]] = None,
+        queries: Optional[List[str]] = None,
         documents: Optional[Union[List[Document], List[List[Document]]]] = None,
-        meta: Optional[Union[Dict[str, Any], List[Dict[str, Any]]]] = None,
-        params: Optional[dict] = None,
-        debug: Optional[bool] = None,
+        invocation_contexts: Optional[List[Dict[str, Any]]] = None,
     ):
-        raise NotImplementedError("run_batch is not implemented for PromptNode.")
+        """
+        Runs PromptNode in batch mode.
+
+        - If you provide a list containing a single query (and/or invocation context)...
+            - ... and a single list of Documents, the query is applied to each Document individually.
+            - ... and a list of lists of Documents, the query is applied to each list of Documents and the results
+              are aggregated per Document list.
+
+        - If you provide a list of multiple queries (and/or multiple invocation contexts)...
+            - ... and a single list of Documents, each query (and/or invocation context) is applied to each Document individually.
+            - ... and a list of lists of Documents, each query (and/or invocation context) is applied to its corresponding list of Documents
+              and the results are aggregated per query-Document pair.
+
+        - If you provide no Documents, then each query (and/or invocation context) is applied directly to the PromptTemplate.
+
+        :param queries: List of queries.
+        :param documents: Single list of Documents or list of lists of Documents in which to search for the answers.
+        :param invocation_contexts: List of invocation contexts.
+        """
+        inputs = self._flatten_inputs(queries, documents, invocation_contexts)
+        all_results: Dict[str, List] = {"results": [], "invocation_contexts": [], "_debug": []}
+        for query, docs, invocation_context in zip(
+            inputs["queries"], inputs["documents"], inputs["invocation_contexts"]
+        ):
+            results = self.run(query=query, documents=docs, invocation_context=invocation_context)[0]
+            output_variable = self.output_variable or "results"
+            all_results[output_variable].append(results[output_variable])
```

Comment:
This will fail if `self.output_variable` is set.

Reply:
Wait I'm a little confused, why would it fail if `self.output_variable` is set? This is also the same logic that is used in the `run` method of `PromptNode`.

Reply:
`all_results` is defined in line 996 with only the following keys: `results`, `invocation_contexts` and `_debug`.

If `self.output_variable` is set to any string that is not one of the three above you'll get a `KeyError` here.

If this is the same logic used in `run` that's broken too and it must be fixed, in a separate PR though.

Reply:
Ahh I see now. This issue is not present in the `run` method. I fixed the error for `run_batch`, set `self.output_variable` in the `__init__` method, and added tests for setting `output_variable` as you suggested.